test/beforeRequest.test.js

```diff
@@ -445,7 +378,7 @@ describe("test validating response", () => {
     describe("test validation and storage", () => {
```

Comment:
So these tests are not done for HC, is this intended? If so, maybe add a comment explaining why.